projects/js-packages/base-styles/_buttons.scss

```diff
@@ -0,0 +1,93 @@
+@import './variables';
+@import './colors';
+
+.components-button {
+	border-radius: var( --jp-border-radius );
+
+	&:focus:not(:disabled) {
+		box-shadow: inset 0 0 0 1px var( --jp-color-white ), 0 0 0 var(--wp-admin-border-width-focus) var( --jp-color-black );
+	}
+
+	&:active:not(:disabled) {
+		color: var( --jp-color-black );
+	}
+}
+
+.components-button.is-primary:not(.is-destructive) {
+	background: var( --jp-color-black );
+
+	&:hover:not(:disabled) {
+		background: var( --jp-color-gray-80 );
+	}
+
+	&:focus:not(:disabled) {
+		box-shadow: inset 0 0 0 1px var( --jp-color-white ), 0 0 0 var(--wp-admin-border-width-focus) var( --jp-color-black );
+	}
+
+	&:disabled {
+		color: rgba( var( --jp-color-white-rgb ), 0.4 );
+		background: var( --jp-color-black );
+	}
+}
+
+.components-button.is-secondary:not(.is-destructive) {
+	color: var( --jp-color-black );
+	box-shadow: inset 0 0 0 1px var(  --jp-color-black );
+
+	&:hover:not(:disabled) {
+		color: var( --jp-color-black );
+		background: var( --jp-color-gray-0 );
+		box-shadow: inset 0 0 0 1px var(  --jp-color-black );
+	}
+
+	&:focus:not(:disabled) {
+		box-shadow: inset 0 0 0 1px var( --jp-color-white ), 0 0 0 var(--wp-admin-border-width-focus) var( --jp-color-black );
+	}
+
+	&:disabled {
+		color: rgba( var( --jp-color-black-rgb ), 0.4 );
+	}
+}
+
+.components-button.is-link {
+	color: var( --jp-color-black );
+
+	&:hover {
+		text-decoration-thickness: 3px;
+	}
+}
+
+.components-button.is-destructive {
+
+
+	&:hover:not(:disabled) {
+		background: var( --jp-color-gray-80 );
+	}
+
+	&:focus:not(:disabled) {
+		color: var( --jp-color-white );
+		box-shadow: inset 0 0 0 1px var( --jp-color-white ), 0 0 0 var(--wp-admin-border-width-focus) var( --jp-color-red-60 );
+	}
+
+	&:disabled {
+		color: rgba( var( --jp-color-white-rgb ), 0.4 );
+	}
+
+	&.is-secondary {
+
+		&:focus:not(:disabled) {
+			color: var( --jp-color-red-50 );
+		}
```

Comment:
One minor issue I noticed.
The button label becomes unreadable on hover, should we make it white?
<img width="267" alt="Screen Shot 2021-11-03 at 3 44 38 PM" src="https://user-images.githubusercontent.com/1341249/140183015-3914a693-3777-4f54-866a-f3206ee74e91.png">

```suggestion
		}
		
		&:disabled {
			color: var( --jp-color-white );
		}
```